cmd/monaco/cmdutils/cmdutils.go

```diff
@@ -26,3 +32,35 @@ func SilenceUsageCommand() func(cmd *cobra.Command, args []string) {
 		cmd.SilenceUsage = true
 	}
 }
+
+// VerifyClusterGen takes a manifestEnvironments map and tries to call the version endpoint of each environment
+// in order to verify that the user has configured the environments correctly.
+// Depending on the configured environment "type" the function tries to call the version endpoint of either
+// 2nd gen cluster (classic) or 3rd gen cluster (platform). The function will return an error as soon as
+// it receives an error from calling the version endpoint of an environment
+func VerifyClusterGen(envs manifest.Environments) error {
+	for _, env := range envs {
+		// Assume 2nd gen cluster and check version endpoint
+		if env.Type == manifest.Classic {
+			if _, err := client.GetDynatraceVersion2ndGen(client.NewTokenAuthClient(env.Auth.Token.Value), env.Url.Value); err != nil {
+				log.Error("Could not verify Dynatrace cluster generation of environment %q (%q). Please check the configured Auth credentials in the manifest", env.Name, env.Url)
+				return fmt.Errorf("unable to call version endpoint of environment %q: %w", env.Name, err)
+			}
+			return nil
+		}
+
+		// Assume 3rd gen cluster an check version endpoint
+		if env.Type == manifest.Platform {
+			oauthCredentials := client.OauthCredentials{
+				ClientID:     env.Auth.OAuth.ClientId.Value,
+				ClientSecret: env.Auth.OAuth.ClientSecret.Value,
+				TokenURL:     "https://sso-dev.dynatracelabs.com/sso/oauth2/token",
+			}
+			if _, err := client.GetDynatraceVersion2ndGen(client.NewOAuthClient(oauthCredentials), env.Url.Value); err != nil {
```

Comment:
I think you meant to call the 3rd gen method here?
```suggestion
			if _, err := client.GetDynatraceVersion3rdGen(client.NewOAuthClient(oauthCredentials), env.Url.Value); err != nil {
```

Reply:
yes I just wanted to test your review skills :D 
I will also forgot to unit test this func